pilot/pkg/config/kube/gateway/conditions.go

```diff
@@ -288,13 +296,25 @@ func generateSupportedKinds(l k8s.Listener) []k8s.RouteGroupKind {
 		intersection := []k8s.RouteGroupKind{}
 		for _, s := range supported {
 			for _, kind := range l.AllowedRoutes.Kinds {
-				if s == kind {
+				if routeGroupKindEqual(s, kind) {
 					intersection = append(intersection, s)
 					break
 				}
 			}
 		}
-		return intersection
+		return intersection, len(intersection) == len(l.AllowedRoutes.Kinds)
+	}
+	return supported, true
+}
+
+// This and the following function really belongs in some gateway-api lib
+func routeGroupKindEqual(rgk1, rgk2 k8s.RouteGroupKind) bool {
+	return rgk1.Kind == rgk2.Kind && getGroup(rgk1) == getGroup(rgk2)
+}
+
+func getGroup(rgk k8s.RouteGroupKind) k8s.Group {
```

Comment:
nit: elsewhere we are using `defaultIfNil((*string)(p.Group), gvk.KubernetesGateway.Group)`. I don't mind having a specific getGroup but we may want to use those primitives (defaultIfNil and the group constant)